app/models/logstash_queries/top_query_match_query.rb

```diff
@@ -36,7 +35,8 @@ def query(json)
   end
 
   def booleans(json)
-    must_affiliate_date_range(json, @affiliate_name, @start_date, @end_date)
+    must_affiliate(json, @affiliate_name)
+    must_date_range(json, @start_date, @end_date)
     must_not_spider(json)
   end
 
```

Comment:
def booleans is the same for top_query_match_query.rb and Date_range_top_n_query.rb .  To apply a more DRY approach you may want to pull this method out and place it in as a concern module that both models can share.

Reply:
@peggles2 , that's similar to how the `DateRangeFilter` module behaved (removed in this commit), combining a couple of the `AnalyticsDsl` methods. I was torn between the DRY-ness of the `DateRangeFilter` module and the simplicity of the `AnalyticsDsl` methods, which each do a simple, single thing to the JSON. After crawling through the tangle of analytics code, I felt it was simpler to have the analytics methods consolidated in one place , `lib/analytics_dsl.rb`. That said, I can be talked out of that approach if you feel this is a devolution. (That code change was not necessary for my story; I was refactoring for clarity.)

Reply:
If its not too much refactoring it be better to have one method handle all of them. Is the DateRangeFilter module very complex? If going to take too much time then we can revisit later.

Reply:
No, it's not too complex. Mostly I was trying to move away from having the analytics methods scattered through a maze of modules. That said, I think I'll revert my refactorings that were not necessary for this story and cross my fingers that someday we'll be able to do a full refactoring of the analytics code.